code/modules/research/experimentor/experimentor.dm

```diff
@@ -0,0 +1,322 @@
+			loaded_item = new/obj/item/unknown_tech/proto_tech()
+			spareparts -= 100
+			ejectItem()
+	src.updateUsrDialog()
+	return
+
+/obj/machinery/r_n_d/experimentor/proc/criticalfailure()
+	switch(rand(1,10))
+		if(1)
+			visible_message("<span class='danger'>Nearby people are hit by debris!</span>")
+			for(var/mob/living/m in oview(1, src))
+				m.apply_damage(15,BRUTE,pick("head","chest","groin"))
+		if(2)
+			visible_message("<span class='danger'>The casing cracks, leaking radiation!</span>")
+			for(var/mob/living/m in oview(1, src))
+				m.apply_effect(25,IRRADIATE)
+		if(3)
+			visible_message("<span class='warning'>The destroyed artifact releases toxic waste!</span>")
+			for(var/turf/T in oview(1, src))
+				if(!T.density)
+					var/obj/effect/decal/cleanable/reagentdecal = new/obj/effect/decal/cleanable/greenglow(T)
+					reagentdecal.reagents.add_reagent("radium", 7)
+		if(4)
+			visible_message("<span class='warning'>The air crackles and pulses with energy!</span>")
+			empulse(src.loc, 4, 0)
+		if(5)
+			var/turf/start = get_turf(src)
+			var/mob/M = locate(/mob/living) in view(src, 3)
+			var/turf/MT = get_turf(M)
+			if(MT)
+				visible_message("<span class='danger'>The coolant cells burst violently!</span>")
+				var/obj/item/projectile/magic/fireball/FB = new /obj/item/projectile/magic/fireball(start)
+				FB.original = MT
+				FB.current = start
+				FB.yo = MT.y - start.y
+				FB.xo = MT.x - start.x
+				FB.fire()
+		if(6)
+			visible_message("<span class='danger'>The damaged device opens a rift in space, vanishing through it!</span>")
+			playsound(src.loc, 'sound/effects/supermatter.ogg', 50, 1, -3)
+			for(var/atom/movable/AM in oview(7,src))
+				if(!AM.anchored)
+					spawn(0)
+						AM.throw_at(src,10,1)
+		if(7)
+			var/mobtype = pick(/mob/living/simple_animal/hostile/bear,/mob/living/simple_animal/hostile/poison/bees,/mob/living/simple_animal/hostile/carp,/mob/living/simple_animal/pet/pug)
+			new mobtype(get_turf(src))
+			visible_message("<span class='danger'>The damaged device opens a rift in space, and something falls out!</span>")
+
+
+
+// Responsible for generating and returning a new functional object to replace the old box.
+/obj/machinery/r_n_d/experimentor/proc/discover(var/obj/item/unknown_tech/T)
+	//Rarity Roll - Starts common. Better tiers overwrite if they crop up.
+	var/rarity = RARITY_COMMON
+	if(prob((T.uncommon_weighting*T.innovation/100+1)*T.uncommon_base))
+		rarity = RARITY_UNCOMMON
+	if(prob((T.rare_weighting*T.innovation/100+1)*T.rare_base))
+		rarity = RARITY_RARE
+	if(prob((T.vrare_weighting*T.innovation/100+1)*T.vrare_base))
+		rarity = RARITY_VERYRARE
+
+	return LootDefiner.define(T.stability, T.potency, T.unpacked_name, rarity, T.containedtype)
+
+#undef SCANTYPE_POKE
+#undef SCANTYPE_IRRADIATE
+#undef SCANTYPE_GAS
+#undef SCANTYPE_HEAT
+#undef SCANTYPE_COLD
+#undef SCANTYPE_OBLITERATE
+#undef SCANTYPE_DISCOVER
+#undef SCANTYPE_RECYCLE
+
+#undef RARITY_COMMON
```

Comment:
Instead of undefing them here I'd suggest making them start with say EXPERIMENTOR_....
This way you don't have to redef them in other files. Might also want to move the defines to the _DEFINE folder since they are used in multiple files